source/leuven/wice_advanced_guide.rst

```diff
@@ -9,15 +9,68 @@ wICE advanced guide
 Compiling software
 ------------------
 
-The wICE compute nodes feature Intel processors nicknamed IceLake, which are
-the successors of the SkyLake and CascadeLake architectures which you can find
-on Genius. Although architectural differences are rather small, it is highly
-recommended to compile a version of your code specifically for wICE. A good approach
-to compile on any cluster (such as wICE) is to launch an interactive job (with the
-``srun`` command).
-
-Many dependencies you might need are centrally installed. The modules that are
-optimized for wICE are available when the appropriate
+Compared to the SkyLake and CascadeLake CPUs on Genius, the wICE nodes
+feature more recent CPU models such as Intel IceLake, Intel Sapphire Rapids
+and AMD Genoa. While architectural differences between SkyLake and CascadeLake
+CPUs can be neglected, the differences with newer CPU models are more
+substantial. When it comes to GPUs there are also significant differences in
+the capabilities of P100, V100, A100 and H100 GPUs.
+
+When locally installing software yourself, we therefore recommend to have
+separate installations for the different types of CPUs and (if applicable)
+GPUs on which you intend to run the software. This rule applies most strongly
+to performance-critical code which is compiled from source (and less strongly
+to light-weight tools based on precompiled binaries).
+
+.. note::
+
+    Remember that precompiled binaries (as is often the case when e.g. Conda
+    or PyPI are involved) are not guaranteed to deliver optimal performance
+    for the target device. In case of doubt, performance-critical parts of
+    an application should not rely on precompiled binaries and instead use
+    optimized binaries as provided by the centrally installed modules and/or
+    by local installations from source. See also :ref:`conda for Python` and
+    :ref:`wice_conda`.
+
+To let jobs use the correct installation at runtime, you can make use of
+predefined environment variables such as ``${VSC_ARCH_LOCAL}`` (and possibly
+``${VSC_ARCH_SUFFIX}`` and ``${VSC_INSTITUTE_CLUSTER}``) to organize your
+installations.
+
+For software using CPUs, the different installations would be:
```

Comment:
I am not sure that the intention here will be clear. Maybe it helps to include a practical example, for instance a job script that shows how to use this?
Maybe it is worth mentioning that on most nodes, `${VSC_ARCH_SUFFIX}` is set to an empty string on most of our nodes?

Reply:
There was indeed still room for improvement there (-> 5bfa41d2c2a706327ab763c179759536d3c6d765)